sdk/servicebus/service-bus/src/core/messageReceiver.ts

```diff
@@ -873,89 +903,74 @@ export class MessageReceiver extends LinkEntity {
         return;
       }
 
-      // We should attempt to reopen only when the receiver(sdk) did not initiate the close
-      let shouldReopen = false;
-      if (receiverError && !wasCloseInitiated) {
-        const translatedError = translate(receiverError) as MessagingError;
-        if (translatedError.retryable) {
-          shouldReopen = true;
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is retryable. This is a candidate for re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        } else {
-          log.error(
-            "[%s] close() method of Receiver '%s' with address '%s' was not called. There " +
-              "was an accompanying error and it is NOT retryable. Hence NOT re-establishing " +
-              "the receiver link.",
-            connectionId,
-            this.name,
-            this.address
-          );
-        }
-      } else if (!wasCloseInitiated) {
-        shouldReopen = true;
+      const translatedError = receiverError ? translate(receiverError) : receiverError;
+
+      // We should only attempt to reopen if either no error was present,
+      // or the error is considered retryable.
+      const shouldReopen =
+        !translatedError || (isMessagingError(translatedError) ? translatedError.retryable : false);
```

Comment:
We changed the `translate()` method quite a bit in Track 2.
Please review that the errors that we would have treated as retryable in Track 1 and still being treated as retryable in Track 2

Reply:
Updating to true as per our discussion.. https://github.com/Azure/azure-sdk-for-js/pull/8580/commits/460e1a51bffd58d7cd739d68daa63768e61f44e9

Reply:
Yes, please do.

Reply:
Because this would affect Event Hubs as well

Reply:
#8653